--- parted-2.2.orig/lib/canonicalize-lgpl.c	2010-02-08 02:19:49.000000000 -0600
+++ parted-2.2.new/lib/canonicalize-lgpl.c	2010-05-08 21:28:21.231262707 -0500
@@ -61,7 +61,9 @@
 #  define __getcwd(buf, max) getwd (buf)
 # endif
 # define __readlink readlink
+#ifndef __set_errno
 # define __set_errno(e) errno = (e)
+#endif
 # ifndef MAXSYMLINKS
 #  ifdef SYMLOOP_MAX
 #   define MAXSYMLINKS SYMLOOP_MAX
